src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/utils/index.ts

```diff
@@ -67,7 +68,13 @@ export const isByte = (type: string): boolean => type.indexOf('byte') === 0
 export const isArrayParameter = (parameter: string): boolean => /(\[\d*])+$/.test(parameter)
 export const getParsedJSONOrArrayFromString = (parameter: string): (string | number)[] | null => {
   try {
-    return JSON.parse(parameter)
+    const arrayResult = JSON.parse(parameter)
+    return arrayResult.map((value) => {
+      if (Number.isInteger(value)) {
+        return new BigNumber(value).toString()
```

Comment:
better use `toFixed`.

The maximum value for uint256 is `2^256 - 1`

`toString` will express big values in scientific notation, while `toFixed` will return the whole number.

```repl
> bn(2**256 - 1).toString()
'1.157920892373162e+77'
> bn(2**256 - 1).toFixed()
'115792089237316200000000000000000000000000000000000000000000000000000000000000'
> 
```

Reply:
Won't the change at line 18 help?

https://github.com/gnosis/safe-react/blob/development/src/index.tsx

Reply:
indeed, it worked :clap: completely forgot about that global setup